src/qtism/data/storage/xml/marshalling/AssessmentResultMarshaller.php

```diff
@@ -0,0 +1,124 @@
+ *
+ * @author Moyon Camille, <camille@taotesting.com>
+ * @license GPLv2
+ */
+
+namespace qtism\data\storage\xml\marshalling;
+
+use DOMElement;
+use qtism\data\QtiComponent;
+use qtism\data\results\AssessmentResult;
+use qtism\data\results\ItemResultCollection;
+
+/**
+ * Class AssessmentResultMarshaller
+ *
+ * The marshaller to manage serialization between QTI component and DOM Element
+ * 
+ * @package qtism\data\storage\xml\marshalling
+ */
+class AssessmentResultMarshaller extends Marshaller
+{
+    /**
+     * Marshall a QtiComponent object into its QTI-XML equivalent.
+     *
+     * @param QtiComponent|AssessmentResult $component A QtiComponent object to marshall.
+     * @return DOMElement A DOMElement object.
+     * @throws MarshallingException
+     */
+    protected function marshall(QtiComponent $component)
+    {
+        $element = self::getDOMCradle()->createElement($this->getExpectedQtiClassName());
+
+        $context = $component->getContext();
+        $element->appendChild($this->getMarshallerFactory()->createMarshaller($context)->marshall($context));
+
+        if ($component->hasTestResult()) {
+            $testResult = $component->getTestResult();
+            $element->appendChild($this->getMarshallerFactory()->createMarshaller($testResult)->marshall($testResult));
+        }
+
+        if ($component->hasItemResults()) {
+            foreach ($component->getItemResults() as $itemResult) {
+                $element->appendChild($this->getMarshallerFactory()->createMarshaller($itemResult)->marshall($itemResult));
+            }
+        }
+
+        return $element;
+    }
+
+    /**
+     * Unmarshall a DOMElement object corresponding to a QTI sessionIdentifier element.
+     *
+     * @param DOMElement $element A DOMElement object.
+     * @return QtiComponent A QtiComponent object.
+     * @throws UnmarshallingException
+     */
+    protected function unmarshall(DOMElement $element)
+    {
+        try {
+            $contextElements = self::getChildElementsByTagName($element, 'context');
+            $contextElement = array_shift($contextElements);
+            $context = $this->getMarshallerFactory()->createMarshaller($contextElement)->unmarshall($contextElement);
+        } catch (\InvalidArgumentException $e) {
+            $msg = "An 'assessmentResult' element must contain one 'context' element, none given.";
+            throw new UnmarshallingException($msg, $element, $e);
+        }
+
+        $assessmentResult = new AssessmentResult($context);
+
+        try {
+            $testResultElements = $element->getElementsByTagName('testResult');
+            if ($testResultElements->length > 0) {
+                $testResultElement = $testResultElements->item(0);
+                $assessmentResult->setTestResult($this->getMarshallerFactory()->createMarshaller($testResultElement)->unmarshall($testResultElement));
```

Comment:
possible bug to`unmarshall($testResultElement));` null pointer exception might occur